packages/mui-material/src/Modal/ModalManager.ts

```diff
@@ -95,23 +114,27 @@ function handleContainer(containerInfo: Container, props: ManagedModalProps) {
     el: HTMLElement | SVGElement;
     value: string;
   }> = [];
-  const container = containerInfo.container;
+
+  // We always try to apply the scrollLock to body in case disablePortal was used
+  // on the modal and container is deeply nested. This method seems to block
+  // scrolling on all containers so it still works with disablePortal
+  const body = ownerDocument(containerInfo.container).body;
```

Comment:
Now that I think of it more I think this break if there are custom overflowing containers provided by the developers through the `container` prop. See https://mui.com/material-ui/api/modal/#modal-prop-container.

Reply:
I will try it in the docs tomorrow, but when I made the parent container overflowing with `disablePortal` enabled the current implementation still worked.

Reply:
I checked this by modifying the docs, it seems fine to me. The page is still not scrollable, you can't scroll the passed-in container either. All modals seems to be fine.

I changed the Menu doc to this:

```typescript
import * as React from 'react';
import Button from '@mui/material/Button';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';

export default function BasicMenu() {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const divRef = React.useRef();
  
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div>
      <div ref={divRef} style={{height: "50px", overflowY: "scroll", display: "flex", flexDirection: "column"}}>
        Hello
        Hello
        <span>Hello</span>
        <span>Hello</span>
        <span>Hello</span>
        <span>Hello</span>
      </div>
      <Button
        id="basic-button"
        aria-controls={open ? 'basic-menu' : undefined}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        onClick={handleClick}
      >
        Dashboard
      </Button>
      <Menu
        id="basic-menu"
        anchorEl={anchorEl}
        container={divRef.current}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
      >
        <MenuItem onClick={handleClose}>Profile</MenuItem>
        <MenuItem onClick={handleClose}>My account</MenuItem>
        <MenuItem onClick={handleClose}>Logout</MenuItem>
      </Menu>
    </div>
  );
}
```